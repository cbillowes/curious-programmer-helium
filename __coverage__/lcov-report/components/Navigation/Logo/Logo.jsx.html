<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Navigation/Logo/Logo.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">components/Navigation/Logo</a> Logo.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react"
import Link from "gatsby-link"
import config from "../../../../data/SiteConfig"
import "./Logo.scss"
&nbsp;
class Logo extends Component {
  render() {
    const title = config.siteTitle.toLowerCase()
    const splitTitle = title.split(" ")
    const left = splitTitle[0]
    const right = splitTitle[1]
&nbsp;
    return (
      &lt;div className="logo"&gt;
        &lt;div className="default-logo"&gt;
          &lt;Link to="/"&gt;
            &amp;#123; &lt;span className="normal-weight"&gt;{left}&lt;/span&gt; {right} &amp;#125;
          &lt;/Link&gt;
        &lt;/div&gt;
        &lt;div className="mobile-logo"&gt;
          &lt;Link to="/"&gt;
            &amp;#123; &amp;#125;
          &lt;/Link&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default Logo</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
