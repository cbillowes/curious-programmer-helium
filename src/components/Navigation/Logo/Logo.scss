@import "../../../layout/variables.scss";

.navigation {
    padding-left: 1em;

    .logo {
        max-width: $container-width;
        margin: 0 auto;
        font-size: 1.5rem;
        padding-top: 5px;

        .mobile-logo {
            display: none;
        }

        .default-logo,
        .mobile-logo {
            margin-bottom: 0.5em;
            font-size: 1.25em;
            font-weight: $font-family-sans-serif-heavy;
            font-family: $font-family-sans-serif;

            a {
                color: $foreground;

                &:hover {
                    color: $primary;
                }
            }
        }
    }
}

@media (max-width: 640px) {
    .navigation {
        .logo {
            .default-logo {
                display: none;
            }

            .mobile-logo {
                display: inline-block;
            }
        }
    }
}
