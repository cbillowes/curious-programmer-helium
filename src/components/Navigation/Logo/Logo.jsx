import React, { Component } from "react"
import Link from "gatsby-link"
import config from "../../../../data/SiteConfig"
import "./Logo.scss"

class Logo extends Component {
  render() {
    const title = config.siteTitle.toLowerCase()
    const splitTitle = title.split(" ")
    const left = splitTitle[0]
    const right = splitTitle[1]

    return (
      <div className="logo" onClick={this.props.onLogoClick.bind(this)}>
        <div className="default-logo">
          <Link to="/">
            &#123; <span className="normal-weight">{left}</span> {right} &#125;
          </Link>
        </div>
        <div className="mobile-logo">
          <Link to="/">
            &#123; &#125;
          </Link>
        </div>
      </div>
    )
  }
}

export default Logo