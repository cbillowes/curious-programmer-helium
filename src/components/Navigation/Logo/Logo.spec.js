import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Logo from "./Logo"

describe("a logo component", () => {
  let element = <Logo />
  let component
  
  beforeAll(() => {
    component = mount(element)
  })

  it("should render", () => {
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render the logo", () => {
    expect(component.find(".logo")).toBeTruthy()
    expect(component.html()).toContain(`href="/"`)
    expect(component.html()).toContain(`{ <span class="normal-weight">curious</span> programmer }`)
  })

  it("should render the mobile logo", () => {
    expect(component.find(".mobile-logo")).toBeTruthy()
    expect(component.html()).toContain(`{ }`)
  })
})