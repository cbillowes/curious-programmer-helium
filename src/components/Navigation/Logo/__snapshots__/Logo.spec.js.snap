// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a logo component should render 1`] = `
<div
  className="logo"
>
  <div
    className="default-logo"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      { 
      <span
        className="normal-weight"
      >
        curious
      </span>
       
      programmer
       }
    </a>
  </div>
  <div
    className="mobile-logo"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      { }
    </a>
  </div>
</div>
`;
